_posts/2023-07-17-elytron-oidc-client-scope.adoc

```diff
@@ -0,0 +1,153 @@
+date: 2023-07-17
+tags: oidc authentication 
+synopsis: How to secure a web application deployed to WildFly with OpenID Connect using additional scope values. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+https://openid.net/developers/how-connect-works/[ OpenID Connect] is an identity layer on top of the OAuth 2.0 protocol that makes it possible for a client to verify a user’s identity based on authentication that’s performed by an OpenID provider. The https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest[OpenID Connect specifications] indicate that there are other scope values which may be included in the authentication request to ask permission to additional and specific resources.The purpose of this example is to demonstrate how to configure additional scope values to request access to additional fields.
+
+toc::[]
+
+Note: This quickstart demonstrates how to configure scope values in two different ways: 
+
+* Deployment Configuration
+* Subsystem Configuration
+
+== Setting up Keycloak OpenID Provider
+
+It’s easy to set up Keycloak using Docker. Follow the steps in Keycloak’s https://www.keycloak.org/getting-started/getting-started-docker[ getting started guide ] to start Keycloak, create a realm called `myrealm` and register a client called `myclient`.
+
+After registering our client, `myclient`, we also need to configure valid redirect URIs. Simply click on `Clients` and then on `myclient`. In the `Valid Redirect URIs field`, enter http://localhost:8090/simple-webapp-oidc/*.
+
+Navigate to the `client scopes` tab for `myclient` and change the `assigned-type` for all scope values from `default` to `optional`. This ensures that you will only get access to claims that you request using the scope values you configure. 
+
+Now, click on Realm roles and create two roles, user and admin. 
+
+Finally, create a user called `alice` whose `First name` is Alice and `Last name` is Smith and assign her the user and admin roles. Steps for assigning roles can be found in the https://www.keycloak.org/docs/latest/server_admin/#proc-assigning-role-mappings_server_administration_guide[ Keycloak documentation]. We will be using different scopes to query these pieces of information about Alice. 
+
+Next, navigate to the `credentials` tab and create a password for Alice. Toggle the `temporary` switch off so you are not prompted to update the password after the first login. The list of scope are the values Keycloak allows and it varies for different OpenID providers.
+
+
+Note: The realm and client names are case sensitive. 
+
+== Server Configuration
+
+First clone the `elytron-example` repo locally: 
+```
+    git clone https://github.com/wildfly-security-incubator/elytron-examples
+    cd elytron-examples
+```
+
+In order to authenticate a user using OpenID connect, we need to configure the server to use OpenID attributes. This can be done using one of the following configurations: 
```

Comment:
In order to secure an application using OpenID Connect, we need to either add configuration in the elytron-oidc-client subsystem or add configuration to the application itself.